### PR TITLE
KiCad: update to 8.0.1

### PR DESCRIPTION
**Pull request by mojca**
#### Description

This is a preliminary attempt to upgrade KiCad to version 8.0.1.

When building 8.0.0 on 10.13 I had to work around a compiler error in json code (which I'm no longer able to reproduce with 8.0.1 on macOS 13, but I no longer have access to 10.13).

This version kind of works for me, but it is possibly problematic as it reverts a bugfix that would require a patched version of wxWidgets. No idea yet what to do with it.

###### Tested on

macOS 10.13.6 17G14042 x86_64
Xcode 10.1 10B61

macOS 13.6.5 22G621 x86_64
Command Line Tools 14.3.1.0.1.1683849156

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/kicad/Portfile [hunk 1]
```diff
@@ -121,6 +124,7 @@ if {${name} eq ${subport}} {
 }
 
 subport kicad-docs {
+    version             8.0.0-rc3
```
Comment (mojca) [inline science/kicad/Portfile:127]: This is taken from a third-party source. Alternative would be to generate documentation from source.

**Comment by pmetzger** [comment]
This seems to have been stuck for some time. Also, 8.0.2 is out now.

**Comment by pmetzger** [comment]
@mojca Should we close this until you can devote more time to it?

**Comment by pmetzger** [comment]
@mojca Poke.

**Comment by mojca** [comment]
Meanwhile the conflict has been introduced by #23321 / https://trac.macports.org/ticket/69647.

The main reason for "not moving forward" is that I was hoping for any kind of feedback or second opinion about this. I could update KiCad to 8.0.2 and resolve conflicts, though I won't have much time to test it in the coming weeks, and I would really like to have some feedback from another member.

We can still use the original wxWidgets as long as we revert one commit (which I already tried, and didn't notice much of a difference).

(As a parallel effort, I started working on upstreaming patches from KiCad to wxWidgets, and the latter are willing to accept them, it's just some work to try to navigate through patches that possibly nobody understands any longer, some that are hardcoded etc. But that "upstreaming patches project" is dormant at the moment.)

**Comment by reneeotten** [comment]
Feel free to reopen once someone had a chance to work / make progress on this!

**Comment by pmetzger** [comment]
@mojca I think there just aren't enough people to review the things you do. Just go ahead and do the right thing and at worst we can update it more later.

**Comment by ok-coder1** [comment]
Hi!
Wanted to check out on how KiCad is doing for 8.0.2 on MacPorts, but it seems like work has stopped?

**Pull request closed without merging**